### PR TITLE
Add caret styling CSS

### LINKED ISSUE #15644: Blinking cursor (almost) not viewable in edit boxes since 1.14
- Gitea version (or commit ref): 1.14

## Description

I'm using Chrome or Edge with forced dark mode (`chrome://flags/#enable-force-dark`) with the arc-green theme.

Blinking cursor is almost invisible in 1.14 it's a very faint cursor (1pixel?) it's fine in 1.13.
This only happens in the edit boxes with the editor, the cursor in the title of the issues etc is ok.

Devtools show it's in the `CodeMirror-scroll` element.

### 1.14
If you squint you can see it ;-)

image

### 1.13

image

### PR DESCRIPTION
**Pull request by silverwind**
Fixes: https://github.com/go-gitea/gitea/issues/15644

Issue is an edge case, but I guess it does not hurt to style the caret explicitely.

### CODE REVIEW

**Comment by 42wim** [comment]
I just build your tree to try it out, but it's not fixed :(

**Comment by 42wim** [comment]
I tried some other iterations with changes in your patch and it only seems to work when this is added to theme-arc-green.less

```
.CodeMirror-cursor {
  border-color: var(--color-text) !important;
}
```

**Comment by silverwind** [comment]
> it only seems to work when this is added to theme-arc-green.less

That would imply some kind of bug with Chrome's dark mode setting in how it deal with CSS variable resolution. I guess we could needlessly copy that CSS rule into arc-green if it really helps.

**Comment by silverwind** [comment]
@42wim try again.

**Comment by 42wim** [comment]
Ok, your patch didn't work and I finally found out why ;-)

In the issue #15644 you told me to fix it by doing 
```
.CodeMirror-cursor {
  border-color: var(--color-text) !important;
}
```
which worked.

But in this PR you're using `--color-caret: var(--color-text-dark);` instead of `--color-caret: var(--color-text);`

By changing this in the `theme-arc-green.less` it works fine now, and also you don't need the double copy of `.CodeMirror-cursor` in  `theme-arc-green.less`

**Review by zeripath — APPROVED** [review]

File: web_src/less/themes/theme-arc-green.less [hunk 1]
```diff
@@ -108,6 +108,12 @@
   --color-placeholder-text: #6a737d;
   --color-editor-line-highlight: var(--color-primary-light-5);
   --color-project-board-bg: var(--color-secondary-light-2);
+  --color-caret: var(--color-text-dark);
```
Comment (42wim) [inline web_src/less/themes/theme-arc-green.less]:
```suggestion
  --color-caret: var(--color-text);
```

**Comment by silverwind** [comment]
Still need to check this because I don't fully understand it yet. There should not be much difference between `--color-text-dark` and `--color-text`.

**Comment by 42wim** [comment]
with `--color-text-dark` the cursor isn't visible, you should be able to reproduce this on edge or chrome with `chrome://flags/#enable-force-dark` or `edge://flags/#enable-force-dark` on enabled.

**Comment by silverwind** [comment]
There seems to be some rather strange contrast algorithm going on with that flag set which seems to invert borders for strange reasons. I would have really liked to use `--color-text-dark` for maximum contrast on that caret but tha bugs out this algorithm leading to Chrome deciding to invert the border. I'm quite surprised that such a buggy feature even made it into Chromium in first place. I guess they are going to drop it at some point.

Anyways, we can land this with the "slightly darker" caret color in code mirror. Proper fix for this can only come with a real `<textarea>`.

**Comment by silverwind** [comment]
Removing backport label as I think we don't need to cater so much to unstable Chromium features like that 😉

**Review by techknowlogick — APPROVED** [review]

**Comment by techknowlogick** [comment]
🚀

**Comment by 42wim** [comment]
thanks @silverwind 👍

**Pull request merged**